File: app/src/main/java/ua/hospes/rtm/domain/race/models/RaceItemDetails.java
package ua.hospes.rtm.domain.race.models;

import android.util.SparseArray;

/**
 * @author Andrew Khloponin
 */
public class RaceItemDetails {
    private int pitStops = 0;
    private SparseArray<Long> completedDriversDuration = new SparseArray<>();


    public RaceItemDetails() {}


    public void addDriverDuration(int driverId, long duration) {
        long oldDuration = getDriverDuration(driverId);
        completedDriversDuration.put(driverId, oldDuration + duration);
    }

    public long getDriverDuration(int driverId) {
        Long l = completedDriversDuration.get(driverId);
        return l == null ? 0L : l;
    }


    public int getPitStops() {
        return pitStops;
    }

    public void setPitStops(int pitStops) {
        this.pitStops = pitStops;
    }


    @Override
    public String toString() {
        return "RaceItemDetails{" +
                "pitStops=" + pitStops +
                ", completedDriversDuration=" + completedDriversDuration +
                '}';
    }
}

File: app/src/main/java/ua/hospes/rtm/domain/race/models/RaceItemDetails.kt
package ua.hospes.rtm.domain.race.models

import android.util.SparseLongArray

data class RaceItemDetails(
        val pitStops: Int = 0
) {
    private val completedDriversDuration = SparseLongArray()

    fun addDriverDuration(driverId: Long, duration: Long) =
            getDriverDuration(driverId).apply { completedDriversDuration.put(driverId.toInt(), this + duration) }.let { Unit }

    fun getDriverDuration(driverId: Long): Long = completedDriversDuration.get(driverId.toInt())
}